collect_app/src/main/java/org/odk/collect/android/javarosawrapper/PropertyManagerJavaRosaInitializer.java

```diff
@@ -0,0 +1,31 @@
+package org.odk.collect.android.javarosawrapper;
+
+import android.app.Application;
+
+import org.odk.collect.android.logic.PropertyManager;
+import org.odk.collect.android.preferences.GeneralSharedPreferences;
+
+import static org.odk.collect.android.logic.PropertyManager.PROPMGR_USERNAME;
+import static org.odk.collect.android.logic.PropertyManager.SCHEME_USERNAME;
+import static org.odk.collect.android.preferences.GeneralKeys.KEY_USERNAME;
+
+public class PropertyManagerJavaRosaInitializer implements JavaRosaInitializer {
+
+    private final Application application;
+
+    public PropertyManagerJavaRosaInitializer(Application application) {
+        this.application = application;
+    }
+
+    @Override
+    public void initialize() {
+        PropertyManager mgr = new PropertyManager(application);
+
+        // Use the server username by default if the metadata username is not defined
```

Comment:
Shouldn't this logic be a responsibility of the `PropertyManager`? Then maybe you don't need this class but can implement the interface inline where needed.

Reply:
Huh yeah I'll play with that.

Reply:
Ended up moving some stuff around so we can just call `reload` on `PropertyManager` and the actual JavaRosa initialization can happen in `ApplicationInitializer`.